text/0000-kebab-actions.md

```diff
@@ -0,0 +1,208 @@
+```
+
+For comparison, to attach the `save` action with classic actions this syntax
+would be used instead:
+
+```hbs
+<button {{action 'save'}}>Save</button>
+```
+
+Kebabs are more verbose in the above case, but have the
+advantage of a clear and extensible syntax. Consider attaching a
+`mouseover` event:
+
+```hbs
+<div on-mouse-over={{action 'save'}}>Save</div>
+```
+
+vs.
+
+```hbs
+<div {{action 'save' on='mouseOver'}}>Save</div>
+```
+
+The kebab version is an basic evolution of the `on-click` usage.
+
+The second advantage of kebab syntax is its parity with Ember component
+action passing. For example consider these components:
+
+```hbs
+{{my-button on-click=(action 'save')}}Save{{/my-button}}
+<my-button on-click={{action 'save'}}>Save</my-button>
+```
+
+Kebab syntax is much closer to the above than classic action syntax. Of course
+a kebab action passed to a component (or angle bracket component) is simply
+a function with no special behavior.
+
+If any non-function is passed to a kebab, such as in
+`on-click="alert('blah');"` an assertion should fail. Only functions are
+permitted.
+
+#### Handling a Kebab Action
+
+Kebab actions will be invoked with the raw DOM event (not the jQuery event)
+as an argument. For example:
+
+```js
+// app/components/my-button.js
+export Ember.Component.extend({
+  actions: {
+    save(event) {
+      console.log('saved with click on ', event.target);
+      this.get('model').save();
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-button.hbs }}
+<button on-click={{action 'save'}}>Save</button>
+```
+
+Actions curry arguments, thus the event may not always be the first argument. For
+example:
+
+```js
+// app/components/my-input.js
+export Ember.Component.extend({
+  actions: {
+    log(prefix, event) {
+      console.log(prefix, Ember.$(event.target).val());
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-input.hbs }}
+<input on-change={{action 'log' 'value is:'}} />
+```
+
+Additionally, the `value` argument to `action` is useful when passing the
+event.
+
+
+```js
+// app/components/my-input.js
+export Ember.Component.extend({
+  actions: {
+    log(value) {
+      console.log(target);
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-input.hbs }}
+<input on-change={{action 'log' value='target.value'}} />
+```
+
+Passing the event is useful for interacting with the originating DOM node,
+but additionally important for allowing event propagation to be controlled
+(cancelled).
+
+#### Event Management
+
+Kebab actions are lazy. If a kebab action of `on-flummux` is used, then Ember
+should listen for the event of `flummux` on the root element and dispatch
+that action when it fires.
+
+If an event from the [list of events Ember listens for](http://emberjs.com/api/classes/Ember.View.html#toc_event-names)
+is used, then kebabs can use the already existing listeners.
+
+# Drawbacks
+
+Obviously, if this is an RFC, there cannot be drawbacks.
```

Comment:
draw back: isn't "just HTML", its a slightly difference from DOM handlers, with a required mental and programmatic translation story. Event delegation + webComponents vs prop based callbacks seems complicated, moving between webcomponents/components/dom elements becomes a refactoring hazard as the micro-syntax changes.

`onclick`, `onClick`, `on-click`